src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -11,14 +11,14 @@
   a name to rename the average column, a number of years to average on
   and the jumpoff year. Returns a dataset where the column to be averaged contains
   the averages for all the years of interest and is renamed to avg-name."
-  [historical-data col-to-avg avg-name number-of-years jumpoff-year]
-  (let [last-yr-data (dec jumpoff-year)
-        start-year-data (inc (- last-yr-data number-of-years))]
-    (-> (i/query-dataset historical-data
-                         {:year {:$gte start-year-data
-                                 :$lte last-yr-data}})
-        (ds/rename-columns {col-to-avg avg-name})
-        (wds/rollup :mean avg-name [:gss-code :sex :age]))))
+  [historical-data col-to-avg avg-name start-yr-avg end-yr-avg]
+  ;; start-yr-range should be >= than the earliest year in historical-data
+  ;; end-yr-range should be <= than the latest year in historical-data
+  (-> (i/query-dataset historical-data
```

Comment:
My point is that validation rules as text, in a comment, are essentially invisible. At the very least, put them in the docstring. Even better would be to validate the values.